pkg/controller/dataimportcron-controller.go

```diff
@@ -673,6 +681,38 @@ func (r *DataImportCronReconciler) createImportDataVolume(ctx context.Context, d
 	return nil
 }
 
+func (r *DataImportCronReconciler) handleStorageClassChange(ctx context.Context, dataImportCron *cdiv1.DataImportCron, desiredStorageClass string) error {
+	digest, ok := dataImportCron.Annotations[AnnSourceDesiredDigest]
+	if !ok {
+		// nothing to delete
+		return nil
+	}
+	name, err := createDvName(dataImportCron.Spec.ManagedDataSource, digest)
+	if err != nil {
+		return err
+	}
+	om := metav1.ObjectMeta{Name: name, Namespace: dataImportCron.Namespace}
+	sources := []client.Object{&snapshotv1.VolumeSnapshot{ObjectMeta: om}, &cdiv1.DataVolume{ObjectMeta: om}, &corev1.PersistentVolumeClaim{ObjectMeta: om}}
+	for _, src := range sources {
+		if err := r.client.Delete(ctx, src); cc.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+	for _, src := range sources {
+		if err := r.client.Get(ctx, client.ObjectKeyFromObject(src), src); err == nil || !k8serrors.IsNotFound(err) {
+			return fmt.Errorf("waiting for old sources to get cleaned up: %w", err)
+		}
+	}
+	// Only update desired storage class once garbage collection went through
+	annPatch := fmt.Sprintf(`[{"op":"add","path":"/metadata/annotations/%s","value":"%s" }]`, strings.ReplaceAll(AnnStorageClass, "/", "~1"), desiredStorageClass)
```

Comment:
check out `EscapeJSONPointer` in apimachinery lib

Reply:
nice ty